File: javascript150/2.ts
class ListNode {
    val: number
    next: ListNode | null
    constructor(val?: number, next?: ListNode | null) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}

function addTwoNumbers(l1: ListNode | null, l2: ListNode | null): ListNode | null {
    let carryOver = 0
    let first = 0
    let answer = new ListNode(0);
    let current = answer

    while (l1 != null || l2 != null) {
        first = l1?.val? l1?.val: 0
        first += l2?.val? l2?.val: 0

        first += carryOver
        carryOver = Math.floor(first/10)
        first %= 10

        
        current.next = new ListNode(first)
        current = current.next

        l1 = l1?.next? l1?.next: null
        l2 = l2?.next? l2?.next: null
    }

    if (carryOver != 0) {
        current.next = new ListNode(carryOver)
    }

    return answer.next
}

File: RandomPractice/2.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carryOver = False

        head = ListNode()
        current = head

        while l1 != None or l2 != None:
            if l1 and l2:
                total = l1.val + l2.val
                if carryOver:
                    total += 1
                    carryOver = False

                if total >= 10:
                    carryOver = True
                    total -= 10

                nextNode = ListNode(total)
                current.next = nextNode
                current = nextNode

                l1 = l1.next
                l2 = l2.next

            elif l1:
                total = l1.val
                if carryOver:
                    total += 1
                    carryOver = False

                if total >= 10:
                    carryOver = True
                    total -= 10

                current.next = ListNode(total)
                current = current.next
            
                l1 = l1.next

            else:
                total = l2.val
                if carryOver:
                    total += 1
                    carryOver = False

                if total >= 10:
                    carryOver = True
                    total -= 10

                current.next = ListNode(total)
                current = current.next
            
                l2 = l2.next

        if carryOver:
            current.next = ListNode(1)
        return head.next
